src/status_im/models/browser-history.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im.models.browser-history
+  (:require [re-frame.core :as re-frame]))
+
+(defn dont-store-history-on-nav-change? [db]
+  (get-in db [:browser/options :dont-store-history-on-nav-change?]))
+
+(defn dont-store-history-on-nav-change []
+  (re-frame/dispatch [:update-browser-options {:dont-store-history-on-nav-change? true}]))
+
+(defn clear-dont-store-history-on-nav-change []
+  (re-frame/dispatch [:update-browser-options {:dont-store-history-on-nav-change? false}]))
+
+(defn dont-store-history-on-nav-change-if-history-exists [db browser-id]
+  (let [browsers (get-in db [:browser/browsers])
+        browser (get browsers browser-id)]
+    (hash-map :dont-store-history-on-nav-change? (some? (:history browser)))))
+
+(defn back [browser]
+  (let [back-index (dec (:history-index browser))
+        back-url (nth (:history browser) back-index)]
+    (dont-store-history-on-nav-change)
+    (re-frame/dispatch [:update-browser (-> browser (assoc :url back-url) (assoc :history-index back-index))])))
+
+(defn forward [browser]
+  (let [forward-index (inc (:history-index browser))
+        forward-url (nth (:history browser) forward-index)]
+    (dont-store-history-on-nav-change)
+    (re-frame/dispatch [:update-browser (-> browser (assoc :url forward-url) (assoc :history-index forward-index))])))
+
+(defn can-go-back? [browser]
+  (not= (:history-index browser) 0))
+
+(defn can-go-forward? [browser]
+  (< (:history-index browser) (dec (count (:history browser)))))
+
+(defn record-history-in-browser-if-needed [db raw-browser url loading]
+  (let [browser (assoc raw-browser :url url)]
+    (cond
+      loading browser
```

Comment:
a newline between the different statements would help making it a bit more readable

Reply:
agree

Reply:
and extra line between conditions

Reply:
Done!